spec/operations/save_operation_spec.cr

```diff
@@ -157,6 +161,92 @@ describe "Avram::SaveOperation" do
     operation.changes[:name].should be_nil
   end
 
+  describe ".find_or_create!" do
+    it "returns the existing record if one exists" do
+      existing_user = UserFactory.create &.name("Rich").nickname(nil).age(20)
+
+      user = UniqueUserSaveOperation.find_or_create!(
+        name: "Rich",
+        nickname: nil,
+        age: 30,
+        joined_at: Time.utc
+      )
+
+      UserQuery.new.select_count.should eq(1)
+      user = user.not_nil!
+      user.should eq(existing_user)
+      # Uses existing age. Does not do an update.
+      user.age.should eq(20)
```

Comment:
This took a while to understand. We're saying find_or_create but it's not using all the values we pass to the method